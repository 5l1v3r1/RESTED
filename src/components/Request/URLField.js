import React, { PropTypes } from 'react';
import { Row, Col, FormGroup, FormControl, ControlLabel } from 'react-bootstrap';

function URLField({ input, meta, placeholderUrl }) {
  return (
    <FormGroup
      controlId="url"
      validationState={meta.touched && meta.invalid ? 'error' : null}
    >
      <Row>
        <Col
          componentClass={ControlLabel}
          sm={2}
        >
          URL
        </Col>

        <Col sm={10}>
          <FormControl
            type="text"
            placeholder={placeholderUrl}
            autoFocus
            {...input}
          />
        </Col>
      </Row>
    </FormGroup>
  );
}

URLField.propTypes = {
  placeholderUrl: PropTypes.string,
  /* eslint-disable react/forbid-prop-types */
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  /* eslint-enable react/forbid-prop-types */
};

export default URLField;

