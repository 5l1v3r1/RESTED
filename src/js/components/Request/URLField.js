import React, { PropTypes } from 'react';
import { Col, FormGroup, FormControl, ControlLabel } from 'react-bootstrap';

function URLField({ input, meta }) {
  return (
    <FormGroup
      controlId="url"
      validationState={meta.invalid ? 'error' : undefined}
    >
      <Col sm={2}>
        <ControlLabel>
          URL
        </ControlLabel>
      </Col>

      <Col sm={10}>
        <FormControl
          type="text"
          {...input}
        />
      </Col>
    </FormGroup>
  );
}

URLField.propTypes = {
  /* eslint-disable react/forbid-prop-types */
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  /* eslint-enable react/forbid-prop-types */
};

export default URLField;

