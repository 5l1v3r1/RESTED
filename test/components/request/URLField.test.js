import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

/* eslint-disable import/no-unresolved */
import URLField from 'components/Request/URLField';

describe('URLField', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      placeholderUrl: 'https://example.com',
      input: {},
      meta: {
        valid: true,
        invalid: false,
      },
    };
  });

  it('should match the previous snapshot', () => {
    const tree = renderer.create(
      <URLField {...initialProps} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match the previous snapshot with invalid field state', () => {
    initialProps.meta.valid = false;
    initialProps.meta.invalid = true;

    const tree = renderer.create(
      <URLField {...initialProps} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an error when the field is invalid', () => {
    initialProps.meta.valid = false;
    initialProps.meta.touched = true;
    initialProps.meta.invalid = true;

    const tree = mount(
      <URLField {...initialProps} />
    );

    expect(tree.find('.form-group').hasClass('has-error')).toBe(true);
  });

  it('should render a placeholder into the field', () => {
    const tree = mount(
      <URLField {...initialProps} />
    );

    expect(tree.find('input').prop('placeholder')).toBe('https://example.com');
  });
});

