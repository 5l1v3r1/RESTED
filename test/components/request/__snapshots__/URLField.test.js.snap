// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URLField should match the previous snapshot 1`] = `
<div
  className="form-group"
>
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    >
      <label
        className="control-label"
        htmlFor="url"
      >
        URL
      </label>
    </div>
    <div
      className="col-sm-10"
    >
      <input
        autoFocus={true}
        className="form-control"
        id="url"
        placeholder="https://example.com"
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`URLField should match the previous snapshot with invalid field state 1`] = `
<div
  className="form-group has-error"
>
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    >
      <label
        className="control-label"
        htmlFor="url"
      >
        URL
      </label>
    </div>
    <div
      className="col-sm-10"
    >
      <input
        autoFocus={true}
        className="form-control"
        id="url"
        placeholder="https://example.com"
        type="text"
      />
    </div>
  </div>
</div>
`;
